drivers/clock_control/nrf5_power_clock.c

```diff
@@ -414,70 +414,32 @@ DEVICE_AND_API_INIT(clock_nrf5_k32src,
 		    &_k32src_clock_control_api);
 
 #if defined(CONFIG_USB) && defined(CONFIG_SOC_NRF52840)
-static void power_int_enable(bool enable)
-{
-	u32_t mask;
-
-	mask = NRF_POWER_INT_USBDETECTED_MASK |
-	       NRF_POWER_INT_USBREMOVED_MASK |
-	       NRF_POWER_INT_USBPWRRDY_MASK;
 
-	if (enable) {
-		nrf_power_int_enable(mask);
-	} else {
-		nrf_power_int_disable(mask);
-	}
-}
-
-static bool usbregstatus_vbusdet_get(void)
+bool nrf5_power_clock_usb_vbusdet(void)
```

Comment:
I see no point in keeping this function. The USB shim can call the POWER HAL itself. Same applies to `nrf5_power_clock_usb_outrdy`.
`nrf5_power_usb_power_int_enable` is a bit different since it refers to the POWER_CLOCK IRQ, so it should stay here (temporarily, until we find a better solution instead of it),